docs/guidelines.md

```diff
@@ -0,0 +1,195 @@
+
+- Format YAML files using [`prettier`](https://prettier.io/) with the default config.
+- Use the `.yml` extension for YAML files (not `.yaml`).
+- Indent using 2 spaces consistently.
+- Add inline comments where logic may not be obvious.
+
+## 4. Workflows
+
+- Repository-specific workflows: `.github/workflows/`
+- Workflow templates for reuse: `.github/workflow-templates/`
+- Reusable workflows: `hemilabs/actions/.github/workflows/`
+
+Refer to [Reusable actions and workflows](#1-reusable-actions-and-workflows) for reusable workflows.
+
+### 4.1. File name
+
+Use lowercase kebab-case file names with the `.yml` extension. Keep names short but descriptive.
+
+The file name should roughly match the workflow name.
+
+**Examples:**
+
+- `go.yml`: Build, lint and test a Go project.
+- `js.yml`: Build, lint and test a JavaScript project.
+- `npm-publish.yml`: Publish an NPM project.
+- `kustomize-deploy.yml`: Deploy using Kustomize.
+
+### 4.2. Workflow Name
+
+Workflow names should be a short, clear label - usually a noun (e.g. "Go" or "JavaScript") - that describes the type of
+project or task the workflow is for.
+
+```yaml
+name: "Go"
+name: "JavaScript"
+name: "Deploy"
+```
+
+Workflow names appear with the job name in the GitHub UI as `<workflow> / <job>`, so both should be readable and
+descriptive together.
+
+**Good examples:**
+
+- `Go / Build`
+- `JavaScript / Lint`
+- `Deploy / Staging`
+
+### 4.3. Jobs
+
+Job names should normally be verbs to describe the action being taken.
+
+**Examples:**
+
+- `Build` - Builds the project.
+- `Test` - Runs tests.
+- `Deploy` - Deploys to an environment.
+- `Lint` - Runs linters.
+- `Publish` - Publish a package.
+
+For context-specific jobs (e.g. when using matrix), consider using parentheses:
+
+- `Build (${{ env.GO_VERSION }})` -> `Go / Build (1.24.2)`
+- `Deploy (staging)`
+
+### 4.5. Steps
+
+Jobs consist of steps, each of which should be:
+
+- Clear in intent.
+- Named descriptively.
+- Ordered logically.
+- Commented if they contain non-obvious logic.
+
+It is good practice to use step names that clearly describe their function, improving maintainability and readability.
+
+Step keys should be ordered as:
+
+- `name`
+- `id`
+- `if`
+- `shell`
+- `working-directory`
+- `env`
+- `uses` / `run`
+- `with`
+
+Ordering keys consistently improves readability and reduces unnecessary diffs during reviews.
+
+```yaml
+- name: "Checkout repository"
+  uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+  with:
+    path: "./example/"
+
+- name: "Read version"
+  id: version
+  run: |
+    VERSION=$(cat version)
+    echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+- name: "Install dependencies"
+  working-directory: "example/"
+  env:
+    GO_VERSION: "1.24.x"
+    TEST_VERSION: "${{ steps.version.outputs.version }}"
+  run: make deps
+```
+
+## 5. Actions
+
+- Repository-specific actions: `.github/actions/`
+- Reusable actions: `hemilabs/actions/`
```

Comment:
Same question as "Reusable workflows" above.

Reply:
Updated to clarify.